### PR TITLE
Add gdscript ts mode

### PR DESCRIPTION
**Pull request by xiliuya**
That supports the use tree-sitter for font-lock, imenu, indentation, and navigation of `gdscript` files.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,35 +10,41 @@ game engine in Emacs. It gives syntax highlighting and indentations.
...
-  - [Adding and removing breakpoints](#adding-and-removing-breakpoints)
-  - [Running the project with the debugger active](#running-the-project-with-the-debugger-active)
-  - [Fetching an object's details](#fetching-an-objects-details)
-  - [Debug Hydra](#debug-hydra)
-  - [The `* Stack frame vars *` buffer](#the--stack-frame-vars--buffer)
-  - [`* Inspector *` buffer](#-inspector--buffer)
-  - [`* Stack dump *` buffer](#-stack-dump--buffer)
-  - [`* Breakpoints *` buffer](#-breakpoints--buffer)
-  - [`* Scene tree *` buffer](#-scene-tree--buffer)
+- [GDScript mode for Emacs](#gdscript-mode-for-emacs)
```
Comment (jcs090218) [inline README.md:15]: Better if we remove the root level? So we keep the same style.

File: gdscript-ts-mode.el [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+;;
+;;; Code:
+
+(when (version< "29" emacs-version)
+  (require 'treesit))
+
+
+
+
```
Comment (jcs090218) [inline gdscript-ts-mode.el:42]: Maybe remove these trailing lines here? WDYT? ;)

**Review by jcs090218 — APPROVED** [review]
Overall LGTM. :)

But I haven't tested since I haven't tried out the treesit.el yet.

File: gdscript-ts-mode.el (hunk 2 above)
Comment (xiliuya) [reply gdscript-ts-mode.el:40]: Yeah, they are.

File: README.md (hunk 1 above)
Comment (xiliuya) [reply README.md]: Hi, @jcs090218 I am not very familiar with the configuration of markdown-doc.

It uses the default format and there are no formatting requirements in this `README.MD` .I have now removed the H1 title. May I ask if there is still a problem

Comment (xiliuya) [reply README.md]: @jcs090218 `markdown-toc` has a bug [issuse/53](https://github.com/ardumont/markdown-toc/issues/53). 
Should I manually edit it?

Comment (xiliuya) [reply README.md]: I have used the following configuration and can now automatically generate directories normally with `markdown-toc`.

```emacs-lisp
(with-eval-after-load 'markdown-toc
  (custom-set-variables '(markdown-toc-user-toc-structure-manipulation-fn 'cdr))
  (setq markdown-toc-indentation-space 2)
  (defun markdown-toc--compute-toc-structure (imenu-index)
    "Given a IMENU-INDEX, compute the TOC structure."
    (--mapcat (markdown-toc--compute-toc-structure-from-level
               (if (eq markdown-toc-user-toc-structure-manipulation-fn 'cdr)
                   -1
                 0)
               it)
              imenu-index)))
```

**Pull request merged**